internal/client/deployment_access.go

```diff
@@ -0,0 +1,100 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+)
+
+var _ = api.DeploymentAccessClient(&DeploymentAccessClient{})
+
+type DeploymentAccessClient struct {
+	hc          *http.Client
+	routePrefix string
+	apiKey      string
+}
+
+// DeploymentAccess returns a DeploymentAccessClient.
+//
+//nolint:ireturn // required to support PrefectClient mocking
+func (c *Client) DeploymentAccess(accountID uuid.UUID, workspaceID uuid.UUID) (api.DeploymentAccessClient, error) {
+	if accountID == uuid.Nil {
+		accountID = c.defaultAccountID
+	}
+
+	if workspaceID == uuid.Nil {
+		workspaceID = c.defaultWorkspaceID
+	}
+
+	return &DeploymentAccessClient{
```

Comment:
wonder if we should have an `IsCloudEndpoint` check and early exit here if no account/workspace ID is provided.  unless we support Deployment ACLs in the OSS server?

https://github.com/PrefectHQ/terraform-provider-prefect/blob/47f66ffd9e4afff2556b663c74a7c8ee2fe45880/internal/client/block_documents.go#L37-L39

Reply:
Ah yes good catch, let me add that - to my knowledge this cloud-only because OSS won't have any concept of a workspace which is a requirement for ACLs to work

Reply:
-> cabc564